compiler/rustc_codegen_llvm/src/base.rs

```diff
@@ -93,6 +94,14 @@ pub fn iter_globals(llmod: &'ll llvm::Module) -> ValueIter<'ll> {
     unsafe { ValueIter { cur: llvm::LLVMGetFirstGlobal(llmod), step: llvm::LLVMGetNextGlobal } }
 }
 
+pub fn svh_symbol_name(tcx: TyCtxt<'_>) -> String {
+    format!(
+        "rust_svh_{}_{}",
+        tcx.original_crate_name(LOCAL_CRATE),
+        tcx.crate_disambiguator(LOCAL_CRATE).to_fingerprint().to_hex()
```

Comment:
This needs to be `crate_hash`. `crate_disambiguator` is passed by cargo using `-Cmetadata` and doesn't depend on the source code.